src/main/typescript/actor/wvActor.ts

```diff
@@ -420,6 +426,40 @@ export default class WvActor extends Actor {
     return updateData;
   }
 
+  async rollCreateCheckMessage(
+    commonData: ChatMessageDataConstructorData,
+    baseFormula: Formulator,
+    target: SerializedCompositeNumber,
+    options: RollOptions | undefined
+  ): Promise<void> {
+    const criticals = this.data.data.secondary.criticals;
+    const fullFormula = baseFormula.modify(options?.modifier).criticals({
+      success: criticals.success.total,
+      failure: criticals.failure.total
+    });
+    const checkRoll = new Roll(fullFormula.toString()).roll({ async: false });
+    await diceSoNice(checkRoll, commonData.whisper ?? null, { actor: this.id });
+    const result = checkRoll.dice[0]?.results[0]?.result ?? 0;
+    const flags: CheckFlags = {
+      type: "roll",
+      details: {
+        criticals, // Is this an automatic type cast? Or is it a result of structural typing?
```

Comment:
Yeah I think this is because of structural typing. You should probably call `toObject()` on all of the Foundry serializable composite numbers.